pkg/reconciler/ingress/controller.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ingress
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/cache"
+
+	network "knative.dev/networking/pkg"
+	"knative.dev/networking/pkg/apis/networking"
+	"knative.dev/networking/pkg/apis/networking/v1alpha1"
+	ingressinformer "knative.dev/networking/pkg/client/injection/informers/networking/v1alpha1/ingress"
+	ingressreconciler "knative.dev/networking/pkg/client/injection/reconciler/networking/v1alpha1/ingress"
+	"knative.dev/networking/pkg/status"
+	endpointsinformer "knative.dev/pkg/client/injection/kube/informers/core/v1/endpoints"
+	"knative.dev/pkg/configmap"
+	"knative.dev/pkg/controller"
+	"knative.dev/pkg/logging"
+	"knative.dev/pkg/reconciler"
+
+	gwapiclient "knative.dev/net-gateway-api/pkg/client/gatewayapi/injection/client"
+	gatewayinformer "knative.dev/net-gateway-api/pkg/client/gatewayapi/injection/informers/apis/v1alpha1/gateway"
+	httprouteinformer "knative.dev/net-gateway-api/pkg/client/gatewayapi/injection/informers/apis/v1alpha1/httproute"
+	"knative.dev/net-gateway-api/pkg/reconciler/ingress/config"
+)
+
+const (
+	// gatewayAPIIngressClassName is the class name to reconcile.
+	gatewayAPIIngressClassName = "gateway-api.ingress.networking.knative.dev"
+)
+
+// NewController initializes the controller and is called by the generated code
+// Registers eventhandlers to enqueue events
+func NewController(
+	ctx context.Context,
+	cmw configmap.Watcher,
+) *controller.Impl {
+	logger := logging.FromContext(ctx)
+
+	ingressInformer := ingressinformer.Get(ctx)
+	httprouteInformer := httprouteinformer.Get(ctx)
+	gatewayInformer := gatewayinformer.Get(ctx)
+	endpointsInformer := endpointsinformer.Get(ctx)
+
+	c := &Reconciler{
+		gwapiclient:     gwapiclient.Get(ctx),
+		httprouteLister: httprouteInformer.Lister(),
+	}
+
+	filterFunc := reconciler.AnnotationFilterFunc(networking.IngressClassAnnotationKey, gatewayAPIIngressClassName, true)
```

Comment:
wondering if we should set it to false to require adding the annotation explicitly.